### PR TITLE
R: make sure hdf5r picks up HDF5 from EB + fix checksum

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb [hunk 1]
```diff
@@ -559,7 +559,7 @@ exts_list = [
         'checksums': ['9b731397b7166dd54941fb0d2eac6df60c7a483b2e790f7eb15b4d7b79c9d69c'],
     }),
     ('boot', '1.3-20', {
-        'checksums': ['f339ddb7138ad05fe3e19852caf7d369f9c1ab827a70045ff66223e032cd0e06'],
+        'checksums': ['adcb90b72409705e3f9c69ea6c15673dcb649b464fed06723fe0930beac5212a'],
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb]: @smoors This reverts the changes done in #8054?

File: easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb [hunk 2]
```diff
@@ -1951,6 +1951,7 @@ exts_list = [
         'checksums': ['58e36d3b8a67d01194c4e5099499657d017f77fc2081213633964b1d57a7e32f'],
     }),
     ('hdf5r', '1.0.0', {
+        'installopts': '--configure-args="--with-hdf5=$EBROOTHDF5/bin/h5pcc"',
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb]: @smoors What goes wrong if this is not used?

File: easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb (hunk 1 above)
Comment (smoors) [reply easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb]: oops..

File: easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb (hunk 2 above)
Comment (smoors) [reply easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb]: if there is a system HDF5 present, `hdf5r` will use that one.

File: easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb (hunk 1 above)
Comment (smoors) [reply easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb]: `curl https://cran.r-project.org/src/contrib/Archive/boot/boot_1.3-20.tar.gz 2>/dev/null | sha256sum`
adcb90b72409705e3f9c69ea6c15673dcb649b464fed06723fe0930beac5212a

File: easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb (hunk 2 above)
Comment (smoors) [reply easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb]: ...and if the system HDF5 is too old, build will fail.

**Comment by boegel** [comment]
I'll take care of this once #8020 is merged (since the same fix for `hdf5r` should be applied in there too)

File: easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/r/R/R-3.5.1-foss-2018b-Python-2.7.15.eb]: Nevermind, it's indeed correct as is, see #8271

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in this PR)
node2001.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/82a6f220c9e7b1c83ed28837f3def108 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2001.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ccd818d6e238eb41a94283d986cb1cb7 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @smoors!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3118.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/5e395d1f9dddc7b0dac5af07132e7203 for a full test report.

**Pull request merged**